apps/studio/components/layouts/TableEditorLayout/ExportAllRows.tsx

```diff
@@ -0,0 +1,431 @@
+
+      const exportResult =
+        params.type === 'provided_rows'
+          ? convertAndDownload(formatRowsForExport(params.rows, params.table), params.table, {
+              convertToOutputFormat,
+              convertToBlob,
+              save,
+            })
+          : await fetchAllRows({
+              queryClient,
+              projectRef: projectRef,
+              connectionString: connectionString,
+              entity: entity,
+              bypassConfirmation,
+              filters: params.filters,
+              sorts: params.sorts,
+              roleImpersonationState: params.roleImpersonationState,
+              progressCallback: totalRows
+                ? (value: number) =>
+                    trackPercentageProgress({
+                      id: entity.id,
+                      name: entity.name,
+                      totalRows: totalRows,
+                      value,
+                    })
+                : undefined,
+              convertToOutputFormat,
+              convertToBlob,
+              save,
+            })
+
+      if (exportResult.status === 'error') {
+        const error = exportResult.error
+        if (error instanceof NoRowsToExportError) {
+          return stopTrackerWithError(
+            entity.id,
+            entity.name,
+            `The table ${entity.name} has no rows to export.`
+          )
+        }
+        if (error instanceof TableTooLargeError) {
+          return stopTrackerWithError(entity.id, entity.name, MAX_EXPORT_ROW_COUNT_MESSAGE)
+        }
+        console.error(
+          `Export All Rows > Error: %s%s%s`,
+          error.message,
+          error.cause?.message ? `\n${error.cause.message}` : '',
+          error.cause?.stack ? `:\n${error.cause.stack}` : ''
+        )
+        return stopTrackerWithError(entity.id, entity.name)
+      }
+
+      if (exportResult.status === 'require_confirmation') {
+        return setConfirmationMessage(exportResult.reason)
+      }
+
+      markTrackerComplete(entity.id, exportResult.rowsExported)
+    }
+  )
+
+  const exportInDesiredFormat = useCallback(
+    () => exportInternal({ bypassConfirmation: false }),
+    [exportInternal]
+  )
+
+  const onConfirmExport = () => {
+    exportInternal({
+      bypassConfirmation: true,
+    })
+    setConfirmationMessage(null)
+  }
+  const onCancelExport = () => {
+    if (!params.enabled) return
+
+    dismissTrackerSilently(params.entity.id)
+    setConfirmationMessage(null)
+  }
+
+  return {
+    exportInDesiredFormat,
+    confirmationModal: confirmationMessage ? (
+      <ConfirmationModal
+        title="Confirm export"
+        visible={true}
+        onCancel={onCancelExport}
+        onConfirm={onConfirmExport}
+      >
+        <div className="flex items-center gap-4">
+          <AlertTriangle aria-label="warning" className="shrink-0 text-foreground-lighter" />
+          <p className="text-sm text-foreground-lighter">{confirmationMessage}</p>
+        </div>
```

Comment:
reckon we use an admonition here for consistency - ConfirmationModal here supports a `alert` param for this